File: backend/src/main/java/com/ingenium/ingeniumecommerce/address/Address.java
package com.ingenium.ingeniumecommerce.address;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;

@Entity
@Table(name = "addresses")
@Builder
@Getter
@NoArgsConstructor
@AllArgsConstructor
public class Address {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String streetName;
    private String streetNumber;
    private String apartmentNumber;
    private String town;
    private String postalCode;
    private String country;
}

File: frontend/src/app/types/address.ts
export interface Address {
    id: number;
    streetName: string;
    streetNumber: string;
    apartmentNumber: string;
    city: string;
    postalCode: string;
    country: string;
}

export interface ShippingAddress {
    firstName: string;
    lastName: string;
    email: string;
    phoneNumber: string;
    streetName: string;
    streetNumber: string;
    apartmentNumber: string;
    city: string;
    postalCode: string;
    country: string;
    paymentType: string;
}
